htdocs/core/db/DoliDB.class.php

```diff
@@ -278,7 +279,9 @@ public function order($sortfield = null, $sortorder = null)
 					$return .= ', ';
 				}
 
+				if ($isnumeric) $return .= 'CAST(';
```

Comment:
Using CAST UNSIGNED may not be compatible with other database like pgsql.
What is the goal of this ?

Reply:
Is INTEGER compatible with MySQL? I get:
```
You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'INTEGER) DESC LIMIT 26' at line 1
```

Reply:
I do not experience trouble on mariadb with 
`select CAST("123" AS INTEGER) from llx_const;`
Which version of mysql do you have ?

Reply:
MySQL 8.0.30
CAST AS INTEGER looks [supported by MariaDB](https://mariadb.com/kb/en/cast/) but [not MySQL](https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html#function_cast).
Should we make a difference according to the RDBMS with INTEGER for everything but MySQL?